pkg/types/azure/machinepool.go

```diff
@@ -43,6 +43,29 @@ type MachinePool struct {
 	// OSImage defines the image to use for the OS.
 	// +optional
 	OSImage OSImage `json:"osImage,omitempty"`
+
+	// securityType specifies the SecurityType of the virtual machine. It has to be set to any specified value to
+	// enable secure boot and vTPM. The default behavior is: secure boot and vTPM will not be enabled unless this property is set.
+	// +kubebuilder:validation:Enum=ConfidentialVM;TrustedLaunch
+	// +optional
+	SecurityType string `json:"securityType,omitempty"`
+
+	// SecureBoot specifies whether secure boot should be enabled on the virtual machine.
+	// Secure Boot verifies the digital signature of all boot components and halts the boot process if
+	// signature verification fails.
+	// If omitted, the platform chooses a default, which is subject to change over time, currently that default is disabled.
+	// +kubebuilder:validation:Enum=Enabled;Disabled
+	// +optional
+	SecureBoot string `json:"secureBoot,omitempty"`
+
+	// VirtualizedTrustedPlatformModule specifies whether vTPM should be enabled on the virtual machine.
+	// When enabled the virtualized trusted platform module measurements are used to create a known good boot integrity policy baseline.
+	// The integrity policy baseline is used for comparison with measurements from subsequent VM boots to determine if anything has changed.
+	// This is required to be set to enabled if the SecurityEncryptionType is defined.
+	// If omitted, the platform chooses a default, which is subject to change over time, currently that default is disabled.
+	// +kubebuilder:validation:Enum=Enabled;Disabled
+	// +optional
+	VirtualizedTrustedPlatformModule string `json:"virtualizedTrustedPlatformModule,omitempty"`
```

Comment:
> In any case, optional fields like this that are added to the installconfig/machine pools need to be pointers. This is due to a dependency on the installer by hive, which needs to serialize install configs that are backwards compatible to various versions. omitempty fields will only be dropped from serialization when they are pointers.

Very useful info, thank you. I didn't figure this out by looking at other `omitempty` fields in this struct, as they are not currently pointers (for example the [EncryptionAtHost boolean](https://github.com/openshift/installer/blob/master/pkg/types/azure/machinepool.go#L21)). I have updated all the optional fields to pointers and adjusted the rest of the code as well.

> Wouldn't it make more sense to have these related fields grouped into their own struct, similar to the way you did [SecuritySettings in the API](https://github.com/openshift/api/blob/master/machine/v1beta1/types_azureprovider.go#L506)?

Great question, I think that you suggested is preferable. I chose this API design initially, because we don't have a VM [SecurityProfile](https://github.com/openshift/api/blob/master/machine/v1beta1/types_azureprovider.go#L491-L502) struct (with the `EncryptionAtHost` and the `SecuritySettings` fields). But that doesn't mean that we shouldn't have these fields grouped (exactly as they are in the respective OpenShift API). I will refactor, test and push the required changes for this grouping.

Reply:
@patrickdillon I have refactored the API to be compatible with the respective OpenShift API and adjusted the rest of the code accordingly. Feel free to take another look whenever possible.